x-pack/plugins/security_solution/server/plugin.ts

```diff
@@ -459,17 +469,23 @@ export class Plugin implements ISecuritySolutionPlugin {
         esClient: core.elasticsearch.client.asInternalUser,
       });
 
-      // Migrate artifacts to fleet and then start the minifest task after that is done
+      // Migrate artifacts to fleet and then start the manifest task after that is done
       plugins.fleet.fleetSetupCompleted().then(() => {
-        logger.info('Dependent plugin setup complete - Starting ManifestTask');
-
         if (this.manifestTask) {
+          logger.info('Dependent plugin setup complete - Starting ManifestTask');
           this.manifestTask.start({
             taskManager,
           });
         } else {
           logger.error(new Error('User artifacts task not available.'));
         }
+
+        turnOffPolicyProtections(
```

Comment:
I think another function name that indicates that turning off the protections is only conditional would be more readable here.

e.g. `turnOffPolicyProtectionsIfNotSupported`, or maybe the check for app features could be moved up here, so `turnOffPolicyProtections()` remains a simple commands:
```ts
if (appFeaturesService.isEnabled(AppFeatureSecurityKey.endpointPolicyProtections)) {
    turnOffPolicyProtections();
}
```

Reply:
My intent here was to not add conditional logic in the `plugin.ts` and just move all of that to the "migration" file. 

I'll rename it as you suggested.